packages/plugin/src/interpreter.ts

```diff
@@ -426,6 +422,15 @@ export class Interpreter {
             assert(DescriptorProto.is(parentDescriptor));
             const ooDecl = parentDescriptor.oneofDecl[fieldDescriptor.oneofIndex];
             info.oneof = this.createTypescriptNameForField(ooDecl);
+            if (localName === this.options.oneofKindDiscriminator) {
+                localName = localName + escapeCharacter;
+            }
```

Comment:
Because of `PartialMessage`, we have to apply this "escaping" to all fields, not just fields in a oneof group.

We can ignore fields with type STRING, and 64-bit integers, because they are not represented as a "string", and therefore will not falsely match the type condition for oneof groups.

I agree that `$kind` is an easy way out, but I'd like to avoid it. I'll think about it.

Reply:
>I agree that `$kind` is an easy way out, but I'd like to avoid it. I'll think about it.

I'm curious why you'd like to avoid it. Is it just aesthetically less pleasing or is there some more practical reason? Changing from `oneofKind` to `$kind` is still fewer keystrokes and on your typically QWERTY keyboard the `$` is very close to the `e` from `one` and you still have to hold `shift`. But yeah, comparing `kind` to `$kind` it's obvious which one is easier to type.

I guess my arguments for `$kind` vs. `kind` are:
* Makes it very obvious that this property isn't coming from your proto file. It's special.
* Avoids the need to complicate the TS/JS property name escaping logic any more than it already is.
* Avoids having to generating TS/JS property names that don't match the proto field names (in this specific instance).